mev-relay-rs/src/service.rs

```diff
@@ -50,7 +50,9 @@ impl Service {
             if let Some(context) = context { context } else { Context::try_from(network)? };
         let clock = context.clock(None);
```

Comment:
not directly related but this would actually improve the code quality here
```suggestion
        let clock = context.clock(Some(genesis_details.genesis_time));
```
 after we move the call to the API up above this line

do you mind doing this? see #80 for some rationale